src/jit/codegencommon.cpp

```diff
@@ -3981,10 +4001,12 @@ void            CodeGen::genFnPrologCalleeRegArgs(regNumber xtraReg,
         {
             // A struct might be passed  partially in XMM register for System V calls.
             // So a single arg might use both register files.
+#ifndef ARM_SOFTFP
             if (isFloatRegType(regType) != doingFloat)
```

Comment:
So to make it clear in genFnPrologCalleeRegArgs() we are getting the local variable values from the incoming registers. isFloatRegType(regType) in the above line really meant that we are working with a floating local variable. In our case we passed the value in an integer arg register, so the original code would just skip such copies at first:
            ```
 if (isFloatRegType(regType) != doingFloat)
             {
                 continue;
             }
```
next it would just try to generate incorrect store or copy instructions using integer regs in the floating context, since it's guided by the type of the local variable which is fp.